### PR TITLE
adding the social security feed

### PR DESCRIPTION
**Pull request by Eran1801**
I added all the feeds of the North, South, etc. of Social Security that published on their website in one script.
Do you prefer that I will separate each region to a diff file feed class and etc.?

### CODE REVIEW

File: libs/feed/social_security_feed.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+from abc import ABC
+from bs4 import BeautifulSoup
```
Comment (liranzxc) [inline libs/feed/social_security_feed.py:2]: Add to requirements.txt all depends libs

Comment (Eran1801) [reply libs/feed/social_security_feed.py:2]: Done @liranzxc

**Review by omercotkd — APPROVED** [review]

File: libs/feed/social_security_feed.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            # from string to json
+            json_data = json.loads(json_string)
+
+            for record in json_data:
+                # adding the state to the record based on the url
+                record['state'] = region
+
+                # edge-case for 'מגידו' that don't have a phone number
+                if record.get('ישוב') == 'מגידו':
+                    continue
```
Comment (liranzxc) [inline libs/feed/social_security_feed.py:65]: why not just filter if record.get(phone) don't have key? maybe tomorrow other cities dont have a phone.

**Review by liranzxc — CHANGES_REQUESTED** [review]

File: apps/carenet-api/requirements.txt [hunk 3]
```diff
@@ -3,4 +3,6 @@ fastapi==0.108.0
 pydantic==2.5.3
 pinecone-client==2.2.4
 sentence-transformers==2.2.2
-python-dotenv==1.0.0
\ No newline at end of file
+python-dotenv==1.0.0
+beautifulsoup4==4.12.3
```
Comment (liranzxc) [inline apps/carenet-api/requirements.txt:7]: the api not using bs4 , the crawler feeds , please add there the requirements.txt

**Review by liranzxc — CHANGES_REQUESTED** [review]

File: libs/feed/social_security_feed.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            # convert table to DataFrame
+            rows = []
+            for tr in table.find_all('tr'):
+                cells = tr.find_all(['th', 'td'])
+                row = [self.clean_text(cell.text) for cell in cells]  # Apply clean_text to each cell's text
+                rows.append(row)
+
+            print(f'The number of documents from the {region} region is: {len(rows) - 1}')
+
+            df = pd.DataFrame(rows[1:], columns=rows[0])
```
Comment (liranzxc) [inline libs/feed/social_security_feed.py:51]: why you need pandas ?

**Review by liranzxc — CHANGES_REQUESTED** [review]

File: libs/feed/social_security_feed.py (hunk 2 above)
Comment (Eran1801) [reply libs/feed/social_security_feed.py:65]: Done @liranzxc

File: apps/carenet-api/requirements.txt (hunk 3 above)
Comment (Eran1801) [reply apps/carenet-api/requirements.txt:7]: Done @liranzxc

File: libs/feed/social_security_feed.py (hunk 4 above)
Comment (Eran1801) [reply libs/feed/social_security_feed.py:51]: For now I will keep it. I wrote to me that I need to take care of it.

**Review by liranzxc — APPROVED** [review]

**Pull request merged**